google/cloud/storage/internal/hash_function_impl_test.cc

```diff
@@ -196,23 +308,43 @@ TEST(HasFunctionImplTest, CreateHashFunctionUpload) {
         ResumableUploadRequest("test-bucket", "test-object")
             .set_multiple_options(test.crc32_disabled, test.crc32_value,
                                   test.md5_disabled, test.md5_value));
-    Update(*function, "The quick brown fox jumps over the lazy dog");
+    function->Update("The quick brown fox jumps over the lazy dog");
     auto const actual = std::move(*function).Finish();
     EXPECT_EQ(test.crc32c_expected, actual.crc32c);
     EXPECT_EQ(test.md5_expected, actual.md5);
   }
 }
 
-TEST(HasFunctionImplTest, CreateHashFunctionUploadResumedSession) {
+TEST(HashFunctionImplTest, CreateHashFunctionUploadResumedSession) {
   auto function = CreateHashFunction(
       ResumableUploadRequest("test-bucket", "test-object")
           .set_multiple_options(UseResumableUploadSession("test-session-id"),
                                 DisableCrc32cChecksum(false),
                                 DisableMD5Hash(false)));
-  Update(*function, "The quick brown fox jumps over the lazy dog");
+  function->Update(kQuickFox);
   auto const actual = std::move(*function).Finish();
   EXPECT_THAT(actual.crc32c, IsEmpty());
-  EXPECT_THAT(actual.crc32c, IsEmpty());
+  EXPECT_THAT(actual.md5, IsEmpty());
+}
+
+TEST(HashFunctionImplTest, CreateHashFunctionInsertObjectMedia) {
+  struct Test {
+    DisableCrc32cChecksum crc32c;
+    DisableMD5Hash md5;
+    HashValues expected;
+  } const cases[] = {
+      {DisableCrc32cChecksum(false), DisableMD5Hash(false),
+       HashValues{kQuickFoxCrc32cChecksum, kQuickFoxMD5Hash}},
+  };
```

Comment:
Did you want to include these 9200 test cases?

https://github.com/googleapis/google-cloud-cpp/blob/74926358837d73d05b8157a138f280cb71e39573/google/cloud/storage/internal/hash_function_impl_test.cc#L260-L304